examples/pytorch/graphsage/graphsage.py

```diff
@@ -0,0 +1,295 @@
+"""
+Inductive Representation Learning on Large Graphs
+Paper: http://papers.nips.cc/paper/6703-inductive-representation-learning-on-large-graphs.pdf
+Code: https://github.com/williamleif/graphsage-simple
+Simple reference implementation of GraphSAGE.
+"""
+import argparse, time, math
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from dgl import DGLGraph
+from dgl.data import register_data_args, load_data
+
+
+def graphsage_msg(edge):
+    msg = edge.src['h']
+    return {'m': msg}
+
+
+class Aggregator(nn.Module):
+    def __init__(self, g, in_feats, out_feats, activation=None, bias=True):
+        super(Aggregator, self).__init__()
+        self.g = g
+        self.weight = nn.Parameter(torch.Tensor(in_feats, out_feats))  # (F,EF)
+        if bias:
+            self.bias = nn.Parameter(torch.Tensor(out_feats))  # (EF,1)
+        else:
+            self.bias = None
+        self.activation = activation
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        weight_stdv = 1. / math.sqrt(self.weight.size(1))
+        self.weight.data.uniform_(-weight_stdv, weight_stdv)
+        if self.bias is not None:
+            bias_stdv = 1. / math.sqrt(self.bias.size(0))
+            self.bias.data.uniform_(-bias_stdv, bias_stdv)
```

Comment:
This procedure can be replaced with `nn.init`.  For example, if you are using ReLU activations, the following runs Xavier uniform initialization with a correction:
```python
        nn.init.xavier_uniform_(self.linear.weight, gain=nn.init.calculate_gain('relu'))
        # if you stick with your code with individual parameters that will be self.weight.
```

Only a hint, since your initialization is different from Xavier uniform.  I saw the original implementation using Xavier uniform though.